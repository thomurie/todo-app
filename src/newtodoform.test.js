import { fireEvent, render } from "@testing-library/react";
import NewTodoForm from "./newtodoform";

// Smoke
test("renders learn react link", () => {
  render(<NewTodoForm />);
});

// Snapshot
test("Matches snapshot", () => {
  const { asFragment } = render(<NewTodoForm />);
  expect(asFragment()).toMatchSnapshot();
});

// Critical Business Logic
test("Can enter data", () => {
  // render the component
  const { getByLabelText } = render(<NewTodoForm />);
  const textInput = getByLabelText("Task Title:");
  // enter data in the form
  fireEvent.change(textInput, { target: { value: "Test Components" } });
  // expect the input data to be in the form
  const inputVal = getByLabelText("Task Title:");
  expect(inputVal.value).toEqual("Test Components");
});
