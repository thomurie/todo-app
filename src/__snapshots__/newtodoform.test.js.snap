// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="NewTodoForm"
  >
    <form
      class="NewTodoForm-Form"
    >
      <label
        for="text"
      >
        Task Title: 
      </label>
      <input
        id="text"
        name="text"
        placeholder="Buy Film"
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Create Task
      </button>
    </form>
  </div>
</DocumentFragment>
`;
